code/modules/surgery/lipoplasty.dm

```diff
@@ -32,20 +33,22 @@
 	name = "remove loose fat"
 	implements = list(TOOL_RETRACTOR = 100, TOOL_SCREWDRIVER = 45, TOOL_WIRECUTTER = 35)
 	time = 32
+	experience_given = 0 //scales with fat removed
 
 /datum/surgery_step/remove_fat/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	display_results(user, target, "<span class='notice'>You begin to extract [target]'s loose fat...</span>",
 			"<span class='notice'>[user] begins to extract [target]'s loose fat!</span>",
 			"<span class='notice'>[user] begins to extract something from [target]'s [target_zone].</span>")
 
-/datum/surgery_step/remove_fat/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results)
+/datum/surgery_step/remove_fat/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
 	display_results(user, target, "<span class='notice'>You extract [target]'s fat.</span>",
 			"<span class='notice'>[user] extracts [target]'s fat!</span>",
 			"<span class='notice'>[user] extracts [target]'s fat!</span>")
 	target.overeatduration = 0 //patient is unfatted
 	var/removednutriment = target.nutrition
 	target.set_nutrition(NUTRITION_LEVEL_WELL_FED)
-	removednutriment -= 450 //whatever was removed goes into the meat
+	removednutriment -= NUTRITION_LEVEL_WELL_FED //whatever was removed goes into the meat
+	experience_given = (round(removednutriment/30))

```

Comment:
> experience_given = (nutrition before surgery - nutrition removed by surgery) / 30

This gives you remaining nutrition if this didnt just set it to well fed. You seem to want the difference, which is (nutrition before surgery - nutrition after surgery).
Can this surgery be done if nutrition is less than well fed? May want to validate the numbers in that case.

Reply:
I want how much was shaved off so whatever their nutrition (L48) was - what it is now (L50 since nutrition was set on 49) should give you that. I might be misunderstanding what you're saying.

You have to be fat to have the surgery but there's prob ways for it to go under so I'll add a check for that.

Reply:
Clumsy wording, but you are currently putting remaining nutrition into the experience calculation instead of the difference.

Reply:
the remaining nutrition should be NUTRITION_LEVEL_WELL_FED per L49 (old L47).

If I had 500 nutrition (new L48) and then I subtract what the nutrition is now which is 450 (new L50), I get 50 which was the amount removed by the surgery. The number I want is the 50, which is why i'm severely confused.

Note we get the nutrition BEFORE setting it to 450, so it should be the 500 in the example (and should give out the 50 when subtracted by 450). Does that make sense?

Reply:
You are right, I was essentially telling you to do what you are already doing. My bad.

Reply:
oh ok rad